evm_loader/program/src/executor.rs

```diff
@@ -149,10 +165,67 @@ impl<'config, B: Backend> Handler for Executor<'config, B> {
         caller: H160,
         scheme: evm::CreateScheme,
         value: U256,
-        init_code: &Vec<u8>,
+        init_code_: &Vec<u8>,
         target_gas: Option<usize>,
     ) -> Capture<(ExitReason, Option<H160>, Vec<u8>), Self::CreateInterrupt> {
-        Capture::Trap(CreateInterrupt{})
+
+        if let Some(depth) = self.state.metadata().depth() {
+            if depth + 1 > self.config.call_stack_limit {
+                return Capture::Exit((ExitError::CallTooDeep.into(), None, Vec::new()));
+            }
+        }
+        // TODO: check
+        // if self.balance(caller) < value {
+        //     return Capture::Exit((ExitError::OutOfFund.into(), None, Vec::new()))
+        // }
+
+        // Get the create address from given scheme.
+        let address =
+            match scheme {
+                evm::CreateScheme::Create2 { caller, code_hash, salt } => {
+                    let mut hasher = Keccak256::new();
+                    hasher.input(&[0xff]);
+                    hasher.input(&caller[..]);
+                    hasher.input(&salt[..]);
+                    hasher.input(&code_hash[..]);
+                    H256::from_slice(hasher.result().as_slice()).into()
+                },
+                evm::CreateScheme::Legacy { caller } => {
+                    let nonce = self.state.basic(caller).nonce;
+                    let mut stream = rlp::RlpStream::new_list(2);
+                    stream.append(&caller);
+                    stream.append(&nonce);
+                    //H256::from_slice(Keccak256::digest(&stream.out()).as_slice()).into()
+                    keccak256_digest(&stream.out()).into()
+                },
+                evm::CreateScheme::Fixed(naddress) => {
+                    naddress
+                },
+            };
```

Comment:
Just for info: this part of code can be placed in the `create_address()` function (like https://github.com/neonlabsorg/evm/blob/c1e3cd05edb6168dc2358d8dca79fcc507b2d3c9/src/executor/stack.rs#L330)